### PR TITLE
Added in turns and teams

### PR DESCRIPTION
**Pull request by Stewart-Taylor**
Added in a simple turn based system and a simple turn system.

Added in gamestate which is also based through to the react render ui function as well.

### CODE REVIEW

**Comment by Stewart-Taylor** [comment]
@cakey

File: src/client/app.coffee [hunk 1]
```diff
@@ -79,6 +81,34 @@ for hexX in [0..12]
 
 scene.add hexagons
 
+
+getAdjacentHexes = ([x, y]) ->
```
Comment (cakey) [inline src/client/app.coffee]: did you mean to add this back in?

File: src/client/app.coffee [hunk 2]
```diff
@@ -79,6 +81,34 @@ for hexX in [0..12]
...
+
+
+class GameState
+    constructor: ->
+        @currentTeamTurn = 0
+        @turn = 1
+
+
+    nextTurn: ->
+        if @currentTeamTurn == 0
```
Comment (cakey) [inline src/client/app.coffee:95]: maybe just (turn % number_of_teams)

File: src/client/app.coffee [hunk 3]
```diff
@@ -95,12 +125,23 @@ class Player
         @mesh.position.x = @x
         @mesh.position.y = @y
 
+    setTeam: (team) ->
+        @team = team
+        if team == 0
+            @material = new THREE.MeshBasicMaterial( { color: "#9b59b6" } )
+            @selectedMaterial = new THREE.MeshBasicMaterial( { color: "#8e44ad" } )
```
Comment (cakey) [inline src/client/app.coffee]: was there a more visible color?

Comment (Stewart-Taylor) [inline src/client/app.coffee]: I changed it to the"Asbetos" just for you

**Comment by cakey** [comment]
@Stewart-Taylor congratz m8

**Pull request merged**